addons/web/static/src/core/datetime/time_picker.js

```diff
@@ -0,0 +1,306 @@
+                        if (navigator.activeItemIndex >= 0) {
+                            this.setValue(this.suggestions[navigator.activeItemIndex]);
+                            this.close();
+                        }
+                    },
+                },
+                arrowdown: {
+                    callback: (navigator) => {
+                        navigator.next();
+                        handleArrow(navigator);
+                    },
+                },
+                arrowup: {
+                    callback: (navigator) => {
+                        navigator.previous();
+                        handleArrow(navigator);
+                    },
+                },
+            },
+        };
+    }
+
+    onPropsUpdated(props) {
+        if (this.suggestions.length === 0) {
+            this.suggestions = this.getSuggestions(props);
+        }
+
+        const getTime = (value) => {
+            if (typeof value === "string") {
+                return parseTime(value, true);
+            } else if (value) {
+                return value;
+            } else {
+                return DateTime.fromObject({ hour: 0, minute: 0, second: 0 });
+            }
+        };
+
+        const newValue = getTime(props.value);
+        if (!timeEquals(newValue, this.lastPropsValue, this.props.showSeconds)) {
+            this.lastPropsValue = newValue;
+            this.lastValue = newValue; // Prevents triggering props.onChange
+            this.setValue(newValue, false);
+        }
+    }
+
+    /**
+     * @param {*} props
+     * @returns {Array<DateTime>}
+     */
+    getSuggestions(props) {
+        const suggestions = [];
+        const rounding = props.minutesRounding <= 5 ? 15 : props.minutesRounding;
+        const minutes = MINUTES.filter((m) => !(m % rounding));
+        for (const hour of HOURS) {
+            for (const minute of minutes) {
+                suggestions.push(DateTime.fromObject({ hour, minute }));
+            }
+        }
+        return suggestions;
+    }
+
+    /**
+     * @param {DateTime} value
+     */
+    setValue(newValue, cleanValue = true) {
+        if (cleanValue) {
+            if (this.props.minutesRounding > 1) {
+                const minute = Math.round(newValue.minute / this.props.minutesRounding) * this.props.minutesRounding;
+                newValue = newValue.set({ minute });
+            }
+            // If showSeconds is false, keep the seconds from
+            // the original props.value
+            if (!this.props.showSeconds && this.state.value) {
+                newValue = newValue.set({ second: this.state.value.second });
+            }
+        }
+
+        this.state.value = newValue;
+        this.state.inputValue = this.timeToString(newValue);
+        this.state.isValid = true;
+
+        if (!timeEquals(this.lastValue, newValue, this.props.showSeconds)) {
+            this.lastValue = copyTime(newValue);
+            this.props.onChange(copyTime(newValue));
```

Comment:
No need to create a helper
```suggestion
            this.lastValue = new DateTime(newValue);
            this.props.onChange(new DateTime(newValue));
```